### DISCUSSION TITLE
안녕하세요 질문있습니다!

### DISCUSSION THREAD

**Discussion by kkimdo20** [tedilabs/fastcampus-devops#56] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/42427215/156019843-add1c458-31d5-475d-b0a8-aa914e90ed20.png)

테라폼으로 환경구축하는 데에서 network 설정하고 ec2 instance에서 terraform init->terraform apply 하는 과정에서 안넘어갑니다.

강제종료하면 SSH authentication failed라고 뜨고, handshake failed라고 하는데 어떤걸 잘못했을까요..??

**Comment by kkimdo20** [comment]
![image](https://user-images.githubusercontent.com/42427215/156030009-719d7faa-5785-4fa5-9027-931cfb42e6ba.png)

타임아웃 된 뒤에는 이렇게 메시지가 뜹니다

**Reply by kkimdo20** [reply]
인스턴스를 중간에 한번 바꿔서 ip 주소가 다르게 캡쳐되었습니다

**Comment by posquit0** [comment, marked as answer]
안녕하세요!
본 이슈는 테라폼을 통해서 EC2 인스턴스를 구동하고 `remote-exec` 프로비저너를 통해서 SSH 명령어를 수행하는 케이스인데요. 테라폼이 SSH 연결에 필요한 인증을 자동으로 수행해주지는 않습니다.
때문에 SSH 연결에 필요한 정보를 사용자가 자신에게 맞는 방법을 선택하여 진행하셔야 합니다.

- `provisioner` 블록 내에 SSH 접속에 필요한 사용자 명 / 패스워드를 기입하시거나 인증서 파일 경로를 기입할 수 있습니다.
- 혹은 SSH Agent를 사용하여 컴퓨터 상에 SSH 인증 정보를 미리 설정해두실 수 있습니다.

### 참고자료

- [Terraform Docs: Provisioners Connection](https://www.terraform.io/language/resources/provisioners/connection)
